src/prefect/flow_runs.py

```diff
@@ -116,15 +118,30 @@ async def main(num_runs: int):
     """
     assert client is not None, "Client injection failed"
     logger = get_logger()
+
+    flow_run = await client.read_flow_run(flow_run_id)
```

Comment:
I actually don't know that this is even necessary to check before we enter the subscriber iterable, maybe we can remove this whole block?

Reply:
hmm I think to avoid any missed events / race conditions due to the time it takes to make this call, we should check for a final state immediately after entering the subscriber (but before the `async for`) , but yeah we should be able to remove that first one outside the subscriber

Reply:
got it, let me know if the new version is what you had in mind